pkg/cmd/infra/deployer/deployer.go

```diff
@@ -196,5 +203,53 @@ func (d *Deployer) Deploy(namespace, deploymentName string) error {
 	} else {
 		glog.Infof("Deploying from %s to %s (replicas: %d)", deployutil.LabelForDeployment(from), deployutil.LabelForDeployment(to), desiredReplicas)
 	}
-	return strategy.Deploy(from, to, desiredReplicas)
+	if err := strategy.Deploy(from, to, desiredReplicas); err != nil {
+		return err
+	}
+	return d.after(config, to)
```

Comment:
Other than it's a breaking change.  I'd say we can do it if someone asks,
but wouldn't do it until they ask.  I just want people to actually give us
feedback on hooks (good or bad) before we make more changes.  Mid is the
only feedback I've gotten, and they didn't even realize they needed it
themselves.

On Mon, Feb 1, 2016 at 11:24 AM, Dan Mace <notifications@github.com> wrote:

> In pkg/cmd/infra/deployer/deployer.go
> <https://github.com/openshift/origin/pull/6925#discussion_r51439932>:
>
> > @@ -196,5 +203,53 @@ func (d *Deployer) Deploy(namespace, deploymentName string) error {
> >  	} else {
> >  		glog.Infof("Deploying from %s to %s (replicas: %d)", deployutil.LabelForDeployment(from), deployutil.LabelForDeployment(to), desiredReplicas)
> >  	}
> > -	return strategy.Deploy(from, to, desiredReplicas)
> > +	if err := strategy.Deploy(from, to, desiredReplicas); err != nil {
> > +		return err
> > +	}
> > +	return d.after(config, to)
>
> Any reason not to make hooks arrays generally (i.e. pre, mid for recreate)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6925/files#r51439932>.
>